lib/run-script-against-resolved-repositories.js

```diff
@@ -0,0 +1,21 @@
+export async function runScriptAgainstRepositories(octokit, repositories, script, options) {
+  for (const repository of repositories) {
+    octokit.log.info(
+      { octoherd: true },
+      "Running on %s ...",
+      repository.full_name
+    );
+
+    try {
+      if (octokit.log.setContext) {
```

Comment:
> Logging for Octoherd is relevant for reports, ...
>
> I think for the programmatic interface we might not need to take any of this into account. I think just using a standard @octokit/core constructor should work, but users of the programmatic interface can pass in their own customized instance of course, e.g. if they want throttling or retries.

i made this conditional on being present, so the reporting behavior is still possible when the octoherd instance of octokit is used. i agree that targeting use with any instance with a base of `@octokit/core` would be ideal for general usage. open to alternative thoughts on approach here if you have any.